code/client/managedsoftwareupdate

```diff
@@ -249,7 +249,8 @@ def doInstallTasks(do_apple_updates, only_unattended=False):
             reports.savereport()
             # install macOS
             try:
-                success = osinstaller.run(finishing_tasks=doFinishingTasks)
+                success = osinstaller.run(finishing_tasks=doFinishingTasks(
+                    runtype='osinstall'))
```

Comment:
The prior version passes the function pointer (a pointer to the doFinishingTasks function). Your change passes the _result_ of _calling_ that function. IOW, `doFinishingTasks(runtype='osinstall')` is called, and its return value is passed to `osinstaller.run()`.  So the "finishing tasks" are run _before_ `osinstaller.run()` is called, and `osinstaller.run()` is being passed the result of a function instead of a function pointer. (Hopefully the result of `doFinishingTasks(runtype='osinstall')` is `None`, or we're likely to get a traceback.)